README.md

```diff
@@ -1,26 +1,178 @@
+
+### Client Credentials
+
+Client credentials allow you to see everything that's publicly available on Vimeo.  This is essentially equivalent to visiting Vimeo.com without signing up for an account or logging in.  This is the simplest authentication method to implement, just one function completes the grant.
+
+```Swift 
+let authenticationController = AuthenticationController(client: vimeoClient)
+
+authenticationController.clientCredentialsGrant { result in 
+	switch result {
+	case .Success(let account):
+		print("Successfully authenticated with account: \(account)")
+	case .Failure(let error):
+		print("error authenticating: \(error)")	
+	}
+}
+```
+
+### Code Grant
+
+If you want to log in as a user, the Vimeo API provides a process called Code Grant authentication.  This is a bit more involved than a client credentials grant, but it gives your application the benefit of making requests on behalf of a Vimeo user, and viewing their personal content.  
+
+To authenticate via code grant, your app launches a specific URL in Safari.  The user signs up or logs in on Vimeo.com, and chooses which permissions to grant.  Then, control is redirected back to your application where authentication completes.
+
+To prepare your application to receive this redirect, navigate to your app target settings > Info > URL Types.  Add a new URL Type, and under url scheme enter `vimeo` followed by your client identifier (for example, if your client identifier is `1234`, enter `vimeo1234`).  You also need to add this redirect URL to your app on the Vimeo API site.  Under “App Callback URL”, add `vimeo{CLIENT_KEY}://auth` (for the example above, `vimeo1234://auth`).
+
+Now, in an appropriate place in your app, open the code grant authorization URL in Safari:
+
+```Swift
+let authenticationController = AuthenticationController(client: vimeoClient)
+
+let URL = authenticationController.codeGrantAuthorizationURL()
+UIApplication.sharedApplication.openURL(URL)
+```
+
+The user will prompted to log in and grant permissions to your application.  When they accept, your redirect URL will be opened, which will reopen your application.  Handle this event in your application delegate:
+
+```Swift
+func application(app: UIApplication, openURL url: NSURL, options: [String : AnyObject]) -> Bool
+    {
+        authenticationController.codeGrant(responseURL: url) { result in
+            switch result {
+            case .Success(let account):
+                print("authenticated successfully: \(account)")
+            case .Failure(let error):
+                print("failure authenticating: \(error)")
+            }
+        }
+        
+        return true
+    }
+```
+
+### Saved Accounts
+
+`AuthenticationController` saves the accounts it successfully authenticates in the Keychain.  The next time your application opens, you should first attempt to load a previously authenticated account before prompting the user to authenticate.
```

Comment:
Should we say "launches" instead of "the next time it opens"? opens

Should we wrap the below logic in applicationDidFinishLaunching to make it clear that launch is where to do this (even if they decide to move it elsewhere)